src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/BasicStatement.scala

```diff
@@ -78,29 +78,28 @@ private[querygeneration] object BasicStatement {
       case l: Literal =>
         l.dataType match {
           case StringType =>
-            val str = l.toString()
-            if (str == "null") {
-              ConstantString(str.toUpperCase) !
+            if (l == null || l.toString() == "null") {
+              ConstantString("NULL") !
             } else {
-              StringVariable(str) ! // else "'" + str + "'"
+              StringVariable(Some(l.toString())) ! // else "'" + str + "'"
             }
           case DateType =>
             ConstantString("DATEADD(day,") + IntVariable(
-              l.value.asInstanceOf[Int]
+              Option(l.value).map(_.asInstanceOf[Int])
             ) +
               ", TO_DATE('1970-01-01'))" // s"DATEADD(day, ${l.value}, TO_DATE('1970-01-01'))"
           case TimestampType =>
             ConstantString("to_timestamp_ntz(") + l.toString() + ", 6)"
```

Comment:
Open question: NULL is considered for before and after CASE, but TimestampType is not changed. Is it intended?

Reply:
I don't understand this question. Could you clarify?

Reply:
What I mean is for before and after CASES, you use Option(l.value).map(_.asInstanceOf[TYPE_NAME]). But, for TimestampType, we don't use this format.
Now, I understand, the before and after, Option(l.value) is needed if asInstanceOf[] is used.
This is resoved.